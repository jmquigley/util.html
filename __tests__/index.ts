import assert from "power-assert";
import {sp} from "util.constants";
import {Fixture} from "util.fixture";
import {newlineToBreak, parseHTML, translateHTML, trimHTML} from "../index";

const debug = require("debug")("util.html::test");

test("Parse a simple HTML file into nodes", () => {
	const fixture = new Fixture("simple");
	assert(fixture);

	const html = fixture.read("test.html");
	assert(html);

	const nodes = parseHTML(html);
	assert(nodes);

	nodes.querySelectorAll("*").forEach((it: HTMLElement) => {
		assert(it);
		expect(it).toMatchSnapshot();
		debug(" -> %O, %o", it, it.nodeName);
	});
});

test("Test translation of HTML string entities", () => {
	assert(translateHTML("abc &quot; def") === 'abc " def');
	assert(translateHTML("abc &nbsp; def") === "abc   def");
	assert(translateHTML("abc &amp; def") === "abc & def");
	assert(translateHTML("abc &lt; def") === "abc < def");
	assert(translateHTML("abc &gt; def") === "abc > def");
});

test("Test special HTML trim function", () => {
	assert(trimHTML(`&nbsp; abc ${sp}`) === "abc");
	assert(trimHTML(`abc &nbsp; def`) === "abc   def");
	assert(trimHTML(`${sp}abc${sp}def${sp}`) === "abc def");
	assert(trimHTML(`&nbsp;abc&nbsp;def&nbsp;`) === "abc def");
});

test("Test changing newlines to the html break tag", () => {
	assert(newlineToBreak(`\ntest\n`) === "<br />test<br />");
	assert(newlineToBreak(`\r\ntest\r\n`) === "<br />test<br />");
	assert(newlineToBreak(`\rtest\r`) === "<br />test<br />");
	assert(newlineToBreak(`\nte\nst\r`) === "<br />te<br />st<br />");
});

test("Test using multiple break strings", () => {
	assert(newlineToBreak("test\n", 2) === "test<br /><br />");
});
