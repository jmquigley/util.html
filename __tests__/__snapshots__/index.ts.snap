// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parse a simple HTML file into nodes 1`] = `
<html
  lang="en"
>
  <head>
    
		
    <meta
      charset="utf-8"
    />
    

		
    <title>
      Simple HTML 5 Template
    </title>
    
		
    <meta
      content="Simple HTML 5 Template"
      name="description"
    />
    
		
    <meta
      content="jmq"
      name="author"
    />
    
	
  </head>
  

	
  <body>
    
		
    <h1>
      Test
    </h1>
    
	


  </body>
</html>
`;

exports[`Parse a simple HTML file into nodes 2`] = `
<head>
  
		
  <meta
    charset="utf-8"
  />
  

		
  <title>
    Simple HTML 5 Template
  </title>
  
		
  <meta
    content="Simple HTML 5 Template"
    name="description"
  />
  
		
  <meta
    content="jmq"
    name="author"
  />
  
	
</head>
`;

exports[`Parse a simple HTML file into nodes 3`] = `
<meta
  charset="utf-8"
/>
`;

exports[`Parse a simple HTML file into nodes 4`] = `
<title>
  Simple HTML 5 Template
</title>
`;

exports[`Parse a simple HTML file into nodes 5`] = `
<meta
  content="Simple HTML 5 Template"
  name="description"
/>
`;

exports[`Parse a simple HTML file into nodes 6`] = `
<meta
  content="jmq"
  name="author"
/>
`;

exports[`Parse a simple HTML file into nodes 7`] = `
<body>
  
		
  <h1>
    Test
  </h1>
  
	


</body>
`;

exports[`Parse a simple HTML file into nodes 8`] = `
<h1>
  Test
</h1>
`;
